src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -1385,6 +1399,9 @@ public void failOnDiskError(final ChecksumFailedException e) {
 
             @Override
             public boolean run(ClientContext context) {
+                synchronized(cooldownLock) {
+                    overallCooldownWakeupTime = -1;
+                }
```

Comment:
This `PersistentJob` is duplicated. Can you break it out in a separate commit please?